### ISSUE TITLE
unable to get an array of callables through static function within a static function

### ISSUE THREAD

**Issue by 1xch** [godotengine/godot#80488]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Ubuntu 20.04.6 LTS (Focal Fossa) - Vulkan (Forward+) - dedicated NVIDIA GeForce GT 710 (nvidia; 470.199.02) 

### Issue description

I am unable to get a static array of callables in a static manner, except the 3rd option below.

Since this is unexpected behavior to me, checking to see if its a bug. I usually use #2, this is the first I've had so much issue I needed to report  

to clarify ALL functions here are prepended with static

```gdscript
# get result by calling CLASS_NAME.a_func_1(v)

# 1
# will run through the array, but the array will be null callables or empty
static func a_func_1(v:Variant) -> String:
	for fn in get_funcs_1:
            fn.call(v)

static var get_funcs_1:Array[Callable] = [
	fn_one,
	fn_two,
	fn_three,
]

# 2
# will NOT run through array, errors at fn_one saying it is nil
static func a_func_2(v:Variant) -> String:
	for fn in get_funcs_2():
            fn.call(v)

static func get_funcs_2 -> Array[Callable]:
	return [
		fn_one,
		fn_two,
		fn_three,
	]

# 3
# this DOES work ONLY if  you create an instance of the class first
# otherwise, the same behavior as #2
static func a_func_3(v:Variant) -> String:
         var funcs_3 = [
		fn_one,
		fn_two,
		fn_three,
	]
	for fn in funcs_3:
            fn.call(v)
```
basically what I usually use in this situation, an empty static class without requiring an instance, is not working as expected so I need clarification on if its a bug, my misunderstanding, or what else is causing the frustration. A class of all consts and static funcs should perform without an instance, to my understanding at this point.

### Steps to reproduce

Put the above in a class, attempt to call CLASS_NAME.a_func_1(v)

### Minimal reproduction project

Not really needed at this point, its relatively simple.

**Comment by Calinou** [comment]
@1xch Did you find a solution to this problem? If so, please document it here by commenting :slightly_smiling_face:

**Comment by 1xch** [comment]
ok. I thought I saw my logic failure, or maybe not. So maybe just reopen it to see what anyone else thinks. 

a_func_2 should be able to call get_funcs_2 which should be able to get static funcs fn_one, fn_two, and fn_three, but  get_funcs_2 can't because its requiring an instance to find the other three.

```
static func a_func_2(v:Variant) -> String:
	for fn in get_funcs_2():
            fn.call(v)

static func get_funcs_2 -> Array[Callable]:
	return [
		fn_one,
		fn_two,
		fn_three,
	]
```
So the question is this my mistake in misunderstanding static function interrelation or a bug. I initially thought that if its all static all the way down, it should not be an issue, as its not in other places: static funcs can call other static funcs so long as they are also static.  

Here's the actual file:
```
# InputEventMatcher
# serialize and deserialize 
class_name IEM
extends RefCounted

const IEM_ACTION:int = 1 << 1			# InputEventAction
const IEM_KEY:int = 1 << 2				# InputEventKey
const IEM_MOUSE_MOVE:int = 1 << 3		# InputEventMouseMotion
const IEM_MOUSE_BUTTON:int = 1 << 4		# InputEventMouseButton
const IEM_JOYPAD_MOVE:int = 1 << 5		# InputEventJoypadMotion
const IEM_JOYPAD_BUTTON:int = 1 << 6	# InputEventJoypadButton
const IEM_SHIFT:int = 1 << 7
const IEM_CTRL:int = 1 << 8
const IEM_ALT:int = 1 << 9
const IEM_META:int = 1 << 10
const IEM_PRESSED:int = 1 << 11
const IEM_RELEASED:int = 1 << 12
const _button_pad:String = "%07d"
const _address_delimiter:String = "."
# const NO_TAG:String = "NO_TAG"

# zero padded 5 digit flag "." zero padded 7 digit button "." string
static func SRZ(e:InputEvent) -> String:
	var acc:Array[String] = []
	var _srz_funcs:Array[Callable] = [
		_e_flagged,
		_e_button,
		_e_tagged,
	]
	for fn in _srz_funcs:
		acc.push_back(fn.call(e))
	return _address_delimiter.join(acc)

#static func _srz_funcs() -> Array[Callable]:
#	return [
#		_e_flagged,
#		_e_button,
#		_e_tagged,
#	]

static func _e_flagged(e:InputEvent) -> String:
	var f:int = 0
	var _flag_funcs:Array[Callable] = [
		_fl_type,
		_fl_modifiers,
		_fl_direction
	]
	for fn in _flag_funcs:
		f = fn.call(e, f)
	return "".num_int64(f)

#static func _flag_funcs() -> Array[Callable]:
#	return [
#		_fl_type,
#		_fl_modifiers,
#		_fl_direction,
#	]

static func _fl_type(e:InputEvent, f:int) -> int:
	if (e is InputEventKey):
		f = _enable(f, IEM_KEY)
		return f
	if (e is InputEventMouseMotion):
		f = _enable(f, IEM_MOUSE_MOVE)
		return f
	if (e is InputEventMouseButton):
		f = _enable(f, IEM_MOUSE_BUTTON)
		return f
	if (e is InputEventJoypadMotion):
		f = _enable(f, IEM_JOYPAD_MOVE)
		return f
	if (e is InputEventJoypadButton):
		f = _enable(f, IEM_JOYPAD_BUTTON)
		return f
	return f

static func _fl_modifiers(e:InputEvent, f:int) -> int:
	if (e is InputEventWithModifiers):
		if e.shift_pressed:
			f = _enable(f, IEM_SHIFT)
		if e.ctrl_pressed:
			f = _enable(f, IEM_CTRL)
		if e.alt_pressed:
			f = _enable(f, IEM_ALT)
		if e.meta_pressed:
			f = _enable(f, IEM_META)
	return f

static func _fl_direction(e:InputEvent, f:int) -> int:
	if e.is_pressed():
		f = _enable(f, IEM_PRESSED)
	if !e.is_pressed():
		f = _enable(f, IEM_RELEASED)
	return f

static func _e_button(e:InputEvent) -> String:
	if (e is InputEventKey):
		return _pads_button(e.keycode)
	if (e is InputEventMouseButton) || (e is InputEventJoypadButton):
		return _pads_button(e.button_index)
	return _pads_button(KEY_UNKNOWN) 

static func _pads_button(v:int) -> String:
	return _button_pad%v

static func _e_tagged(e:InputEvent) -> String:
	if (e is InputEventAction):
		return e.action
	return "" # NO_TAG # e.as_text().sha256_text()
	
static func DSRZ(v:String) -> InputEvent:
	# var spl = v.split(_address_delimiter)
	return null

## flag

static func _checks(v:int, f:int) -> bool:
	return v & f != 0

static func _enable(v:int, f:int) -> int:
	return v|f

static func _disable(v:int, f:int) -> int:
	return v & ~f
```

**Comment by 1xch** [comment]
My actual solution:

```
static func a_func_2(v:Variant) -> String:
	for fn in get_funcs_2():
            fn.call(v)

static func get_funcs_2 -> Array[Callable]:
	return [
		CLASS_NAME.fn_one,
		CLASS_NAME.fn_two,
		CLASS_NAME.fn_three,
	]
```
